### DISCUSSION TITLE
SSRF issue dependency of @inertiajs/inertia [dev]

### DISCUSSION THREAD

**Discussion by devlloedy** [inertiajs/inertia#402] [Help] [answered]
Hi! I got this issue (See picture below) pertaining to this issue https://www.npmjs.com/advisories/1594/versions . 
I npm install to resolve the vulnerability but now my site became broken

![image](https://user-images.githubusercontent.com/75713279/103615174-12c31c80-4f65-11eb-801a-7ff9cc178475.png)

console error
![image](https://user-images.githubusercontent.com/75713279/103615364-6df50f00-4f65-11eb-98d2-a802fbc4c725.png)

how can i fix this? this is my first time to encounter this error. if i revert back in old version my site works fine thanks.

**Comment by claudiodekker** [comment]
Hi,

To fix this issue, both the `@inertiajs/inertia` package as well as the adapter package you're using (e.g. `@inertiajs/inertia-vue`) should be compatible. Here's a list of all compatible adapter packages for `@inertiajs/inertia`:

- https://github.com/inertiajs/inertia/releases/tag/inertia-vue%40v0.5.4
- https://github.com/inertiajs/inertia/releases/tag/inertia-vue3%40v0.3.4
- https://github.com/inertiajs/inertia/releases/tag/inertia-react%40v0.5.1
- https://github.com/inertiajs/inertia/releases/tag/inertia-svelte%40v0.6.0

Do note that when making an upgrade like this, there are things in your application that might break due to changes between the versions. As such, I recommend you to go over to the releases page (https://github.com/inertiajs/inertia/releases) and to go through the releases all the way from your application's installed versions up to the current ones, applying all of the breaking changes/fixes that are relevant to your application. You can ignore all packages on the release page that aren't relevant for you.

In addition, I'd like to take this opportunity to mention that we understand that this isn't ideal, but we do intend to make 'breaking changes' like this less frequent once version 1.0.0 gets released. Furthermore, there is a serious consideration of removing Axios from Inertia's dependencies altogether, so that you're free to upgrade (or remove) it separately without having to go through the hassle of being (more or less) forced to upgrade your entire Inertia app.

Hope this helps!

**Comment by reinink** [comment]
Also, make sure you're not on `axios` version `0.21.0`, as there is a [known vulnerability](https://github.com/advisories/GHSA-4w2v-q235-vp99). Upgrade to `0.21.1`. 👍

**Comment by reinink** [comment, marked as answer]
I just opened a PR to force this: https://github.com/inertiajs/inertia/pull/409
